client/app/pages/queries/QuerySource.jsx

```diff
@@ -239,9 +244,23 @@ function QuerySource(props) {
     }
   }, []);
 
-  const canExecuteQuery = true; // TODO: Replace with real value
+  const canExecuteQuery = useMemo(
+    () =>
+      !query.getParameters().hasPendingValues() &&
```

Comment:
This doesn't seem to work :confused:, for the view page I used a state with this Parameters callback:
https://github.com/getredash/redash/blob/dd34a2bf233d6a354f9cd040ea746528da22fd09/client/app/pages/queries/QueryView.jsx#L129

Reply:
Thank you! Fixed